tools/Cron_jobs/quarantine_report.php

```diff
@@ -315,6 +315,27 @@ function return_quarantine_list_array($to_address, $to_domain)
     return $array;
 }
 
+//ASU MODIFIED CODE BEGIN
+function get_random_string()
+{
+    $bytes = openssl_random_pseudo_bytes(10);
+    return bin2hex($bytes);
+}
+
+function store_auto_release($qitem)
+{
+    $id = $qitem['id'];
+    $rand = $qitem['rand'];
+    $result = dbquery("INSERT INTO mod_release (msg_id,uid) VALUES ('$id','$rand')");
+    if (!$result) {
+        dbg(" ==== Error generating auto_release....skipping...");
+        return false;
+    } else {
+        return true;
```

Comment:
The generated random value is stored in $qitem['rand'], for use in the email link